mlir/include/mlir/Dialect/Rock/IR/RockOps.td

```diff
@@ -50,6 +50,12 @@ class TensorOrMemRefRankOf<list<Type> allowedTypes, list<int> ranks> :
              /*summary=*/"",
              "::mlir::ShapedType">;
 
+class TensorOrMemRefOf<list<Type> allowedTypes> :
+  AnyTypeOf<[TensorOf<allowedTypes>,
+             MemRefOf<allowedTypes>],
+             /*summary=*/"",
```

Comment:
Can you make the `summary` field meaningful? We'll want that for error messages to work.

Reply:
Done!